Bug fixes, clean up, and upgrades

Protonull opened the pull request:
This does not update the plugin to using CivModCore, that will be done at a later stage. This is instead a list of general improvements.

 1. `/iec input` and `/iec output` and be shortened to `/iec in`, `/iec i` and `/iec out`, `/iec o` respectively.

 2. Splash Potions and Lingering Potions now display correctly in the shop GUI

 3. Cleaned up old code and comments that convey a lack of support for potions

 4. Coloured display names will no longer bleed out of their quotations.

 5. Improved the materials.csv and enchantments.csv to bring them up to date.

 6. Fixed the bug preventing players from creating exchange rules for Water Bottles, Awkward Potions, and other potion types without effects.

 7. Fixed item matching so exclusive and excluded enchantments are now respected.

Protonull committed:
Expanded RuleType versions.

You can now type `/iec in` or `/iec i` instead of just `/iec input`, and similar versions for output.

Protonull committed:
Fixed a visual error for splash and lingering potions.

While splash and lingering potions still could be traded (ExchangeRule.parseItemStack() recognises potions via Metadata instanceof checks and not Material matching), they'd not have the same GUI output as regular potions.

Protonull committed:
Removed `POTION` from the `NOT_SUPPORTED` list.

Protonull committed:
Fixed display name bug.

If a display name contained a colour, that colour would bleed out past the name.

Protonull committed:
Update to the materials list.

A similar update will be coming to my fork of CivModCore, but as ItemExchange does not use CivModCore.

Protonull committed:
Optional addition display.

Protonull committed:
Adds default potion base if all else fails.

Protonull committed:
Fixed effectless potion bug.

PotionMetadata can now account for potions that do not produce effects, such as Water Bottles and Akward Potions.

Protonull committed:
Brought enchantments.csv up to date

Protonull committed:
Typo

Protonull committed:
Upgraded the item matching function.

Excluded and exclusive enchantments are now respected.

Maxopoly commented on src/main/java/com/untamedears/ItemExchange/utility/ExchangeRule.java:
```diff
@@ -545,47 +548,79 @@ public int checkMultiples(Inventory inventory) {
-			for (Enchantment excludedEnchantment : excludedEnchantments) {
-				followsRules = followsRules && !itemStack.getEnchantments().entrySet().contains(excludedEnchantment);
+		// If not the same material, return false
+		if (DeprecatedMethods.getMaterialId(material) != DeprecatedMethods.getItemId(itemStack)) {
```
Could have just done `getType` on the ItemStack and then compare materials via `==`, no ?

Protonull replied:
Yes actually, my bad

Maxopoly commented on src/main/java/com/untamedears/ItemExchange/utility/ExchangeRule.java:
```diff
@@ -545,47 +548,79 @@ public int checkMultiples(Inventory inventory) {
+		}
+		// Check required enchantments
+		Map<Enchantment, Integer> itemEnchants = itemStack.getEnchantments();
+		if (itemEnchants.isEmpty() != requiredEnchantments.isEmpty()) {
```
Everything from here to line 574 could be simplified to something like:

```
if (!unlistedEnchantmentsAllowed && itemEnchants.size() != requiredEnchantments.size()) {
				return false;
}
if (!itemEnchants.entrySet().containsAll(requiredEnchantments.entrySet())) {
				return false;
}

```

Protonull replied:
I wasn't sure if comparing two empty collections in a .containsAll() would return true or false. It turns out that an empty collection _does_ contain all the elements from another empty collection. Huh, TIL. But it means I can do the simplification you suggested

Maxopoly commented:
Minor complaints, aside from that it looks good. I assume you tested it?

Protonull committed:
Suggested changes

Protonull committed:
Added shortened /iec options

More involved /iec commands are handled outside of the CreateCommand class, and so the expanded options are copied here too
